src/test.py

```diff
@@ -62,10 +64,63 @@ def wrap_socket(self, s, server_hostname):
     def patch(cls):
         return mock.patch("ssl.create_default_context", wraps=cls)
 
+class SilentTk:
+    def bind(self, event, callback):
+        pass
+
+tkinter.Tk = SilentTk
+
+class SilentCanvas:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_text(self, x, y, text):
+        pass
+
+    def pack(self):
+        pass
+
+    def delete(self, v):
+        pass
+
+tkinter.Canvas = SilentCanvas
+
+class MockCanvas:
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def create_text(self, x, y, text):
+        print("create_text: x={} y={} text={}".format(x, y, text))
+
+    def pack(self):
+        pass
+
+    def delete(self, v):
+        pass
+
+original_tkinter_canvas = tkinter.Canvas
+
```

Comment:
Shouldn't this come before we reassign it?

Reply:
"original" means SilentCanvas. I found that regular tkinter doesn't work in the unittest framework, because the environment is not real enough.